gramps/gen/filters/rules/repository/_hasrepo.py

```diff
@@ -67,10 +67,12 @@ def prepare(self, db, user):
             self.rtype = RepositoryType()
             self.rtype.set_from_xml_str(self.list[1])
 
-    def apply(self, _db, obj):
+    def apply_to_one(self, _db, data):
         """
         Apply the rule. Return True on a match.
         """
+        obj = self.get_object(data)
+        # FIXME: probably can do the rest without object:
         if not self.match_substring(0, obj.get_name()):
```

Comment:
```suggestion
        if not self.match_substring(0, data["name"]):
```

Reply:
Resolved to `obj.name` with new attr accessor.